### ISSUE TITLE
[Most Wanted] Can't use XINPUT_GAMEPAD_RT as GAMEACTION_NOS and XINPUT_GAMEPAD_LT as GAMEACTION_SHIFTUP

### ISSUE THREAD

**Issue by JubilantNebula** [xan1242/NFS-XtendedInput#74]
PC:
OS: Windows 10

NFSMW:
XtendedInput Version: 1.20

Default: No problem (Can GAMEACTION_GAS w/ XINPUT_GAMEPAD_RT and GAMEACTION_BRAKE w/ XINPUT_GAMEPAD_LT)
Custom: Problem w/ XINPUT_GAMEPAD_RT and XINPUT_GAMEPAD_LT

Cfg Modified:
```
; In-Game controls
GAMEACTION_GAS = XINPUT_GAMEPAD_B
GAMEACTION_BRAKE = XINPUT_GAMEPAD_A
GAMEACTION_STEERLEFT = XINPUT_GAMEPAD_LS_LEFT
GAMEACTION_STEERRIGHT = XINPUT_GAMEPAD_LS_RIGHT
GAMEACTION_TURNLEFT = XINPUT_GAMEPAD_LS_LEFT
GAMEACTION_TURNRIGHT = XINPUT_GAMEPAD_LS_RIGHT
GAMEACTION_HANDBRAKE = XINPUT_GAMEPAD_X
; Here!!
GAMEACTION_NOS = XINPUT_GAMEPAD_RT
GAMEACTION_GAMEBREAKER = XINPUT_GAMEPAD_RB
;; and Here!!
GAMEACTION_SHIFTUP = XINPUT_GAMEPAD_LT
GAMEACTION_SHIFTDOWN = XINPUT_GAMEPAD_LB
GAMEACTION_RESET = XINPUT_GAMEPAD_BACK
HUDACTION_PAUSEREQUEST = XINPUT_GAMEPAD_START
HUDACTION_ENGAGE_EVENT = XINPUT_GAMEPAD_DPAD_UP
HUDACTION_PAD_LEFT = XINPUT_GAMEPAD_DPAD_LEFT
HUDACTION_PAD_DOWN = XINPUT_GAMEPAD_DPAD_DOWN
HUDACTION_PAD_RIGHT = XINPUT_GAMEPAD_DPAD_RIGHT
HUDACTION_SKIPNIS = XINPUT_GAMEPAD_A
HUDACTION_NEXTSONG = XINPUT_GAMEPAD_LEFT_THUMB
CAMERAACTION_CHANGE = XINPUT_GAMEPAD_RIGHT_THUMB
CAMERAACTION_LOOKBACK = XINPUT_GAMEPAD_Y
```

**Comment by diab0l** [comment]
# TL;DR

in `scripts/NFS_XtendedInput.ini`
change `Percent_AnalogTriggerDigital` to `0`

# I LIKE TO READ

I've had the same problem playing in wine with a wired xbox 360 controller.
So, I've looked through the sources and couldn't find any obvious bug.
It should work (tm), but it doesn't.

My guess is that the scaling for the triggers is wrong such that even when they are fully pressed, there is only a tiny activation and that tiny activation is interpreted as the trigger moving into the deadzone by default :D
Changing the deadzone to a much smaller value (or just 0) fixes the problem.

Ideally somebody should figure out why the scaling is so wrong for the triggers, currently the value is somewhere above 0 and below 0.1 when the trigger is fully pressed.

**Comment by kamik-hub** [comment]
> in `scripts/NFS_XtendedInput.ini`
> change `Percent_AnalogTriggerDigital` to `0`

Percent_AnalogTriggerDigital = 0 and Percent_Shifting = 0 fixed this problem, thanks
